### ISSUE TITLE
Compile shaders when using Zig package manager

### ISSUE THREAD

**Issue by tiawl** [Snektron/vulkan-zig#104]
Hi,

Reading the `README.md` I found 2 sections about how to use `vulkan-zig` with the Zig package manager. So I started to script something with this idea in mind. Unfortunately when it came to compile my shaders, I had the feeling that the `README.md` section concerning shader compilation was not written for this use case. Maybe I am missing something but, glancing your code, I litterally have no idea how to solve my issue. Would you consider adding this feature or explaining what I am missing ?

Thank you.

**Comment by antlilja** [comment]
You can import vulkan-zig as a build dependency and use the ```ShaderCompileStep``` to compile your shaders like is done for the [example](https://github.com/Snektron/vulkan-zig/blob/510c6b836b7b016572c43bd980abf4d643fdf066/build.zig#L56C5-L56C5).

Here's how you can import vulkan-zig as a build dependency and use the ```ShaderCompileStep```:
```zig
const std = @import("std");
const ShaderCompileStep = @import("vulkan-zig").ShaderCompileStep;

pub fn build(b: *std.Build) !void {
    const target = b.standardTargetOptions(.{});
    const optimize = b.standardOptimizeOption(.{});

    const spec_path: []const u8 = b.pathFromRoot("vk.xml");
    const vz_dep = b.dependency("vulkan-zig", .{ .registry = spec_path });

    const exe = b.addExecutable(.{
        .name = "app",
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });
    exe.addModule("vulkan", vz_dep.module("vulkan-zig"));
    b.installArtifact(exe);

    const shaders = ShaderCompileStep.create(
        b,
        &[_][]const u8{ "glslc", "--target-env=vulkan1.2" },
        "-o",
    );
    shaders.add("vert", "shader.vert", .{});
    shaders.add("frag", "shader.frag", .{});
    exe.addModule("shaders", shaders.getModule());
}
```

**Comment by tiawl** [comment]
Hi,

First of all, thank you for your answer.

Maybe I am missing something but your `build.zig` is not working or it could work in a way it does not satisfy my need.

Let me explain: you are pointing the `vulkan-zig/build.zig` file. You are right this build file is working. But it works because it is inside the repository itself. So it can access other files declarations with an `@import ()`. So my `build.zig` could work if I clone `vulkan-zig` in my workspace and I use it like it is done in `vulkan-zig/build.zig`. But that is not what I want. I do not want to clone the `vulkan-zig` repository if I am using the Zig Package Manager to get the `vulkan-zig` repository.

Maybe rewording my request in a simple question will help: how do I access to a `ShaderCompileStep` object through a `std.Build.Dependency` object ?

**Comment by antlilja** [comment]
> Hi,
> 
> First of all, thank you for your answer.
> 
> Maybe I am missing something but your `build.zig` is not working or it could work in a way it does not satisfy my need.
> 
> Let me explain: you are pointing the `vulkan-zig/build.zig` file. You are right this build file is working. But it works because it is inside the repository itself. So it can access other files declarations with an `@import ()`. So my `build.zig` could work if I clone `vulkan-zig` in my workspace and I use it like it is done in `vulkan-zig/build.zig`. But that is not what I want. I do not want to clone the `vulkan-zig` repository if I am using the Zig Package Manager to get the `vulkan-zig` repository.
> 
> Maybe rewording my request in a simple question will help: how do I access to a `ShaderCompileStep` object through a `std.Build.Dependency` object ?

You don't need to clone vulkan-zig into your workspace in order to import it. You can import the build.zig if you have vulkan-zig present in your ```build.zig.zon``` file.

It should work if you use the above example with this ```build.zig.zon```file:

```
.{
	.name = "app",
	.version = "1.0.0",

	.dependencies = .{
		.@"vulkan-zig" = .{
			.url = "https://github.com/Snektron/vulkan-zig/archive/510c6b836b7b016572c43bd980abf4d643fdf066.tar.gz",
			.hash = "12205fcba4cd4d81ac268810e136218ed4a2eeef49f185b23fb37a71cabcc62d5a6a",
		},
	},
}
```

**Comment by tiawl** [comment]
Ok the problem was coming from my `build.zig.zon` and the `.hash` member. My build file did not find the `ShaderCompileStep` because this field was not correctly set. Thank you for your help and sorry to bother for this ridiculous mistake.
